### DISCUSSION TITLE
Interactive shell using Docker Compose

### DISCUSSION THREAD

**Discussion by naveedalirehmani** [community/community#63697] [Programming Help]
### Body

Can I start a live shell in a container using just Docker Compose? I tried using /bin/sh as the entrypoint in my docker-compose file, but the container stops right away. Is there a way to keep it running interactively using Docker Compose? I'm aware of native docker commands, but I want to know if it can be done with Docker Compose alone.

```
myapp:
  image: alpine:latest
  entrypoint: /bin/sh
```

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by pervaizdattoo** [comment]
In the provided example from the official Docker Compose "Getting Started" guide (accessible at https://docs.docker.com/compose/gettingstarted/), the accompanying docker-compose.yml configuration is as follows:

`version: "3.9"
services:
  web:
    build: .
    ports:
      - "8000:5000"
  redis:
    image: "redis:alpine"`
    
    
    Once you initiate this setup using the command docker-compose up, you have the option to access a shell within either the redis container or the web container using the following commands:

To access the shell in the redis container:

`docker-compose exec redis sh
`

To access the shell in the web container:

`docker-compose exec web sh`
